src/pages/index.js

```diff
@@ -23,45 +34,172 @@ import {
   editPopupSelector,
   nameSelector,
   jobSelector,
-  initialCards,
   cardWrapperSelector,
 } from "../utils/constants.js";
 import UserInfo from "../components/UserInfo.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
+// elements
+let cardList;
+const deleteCardPopup = new PopupWithConfirmation(deleteCardSelector);
+const imageInfo = new PopupWithImage(picPopupSelector);
+const userProfile = new UserInfo({ nameSelector, jobSelector, avatarSelector });
+const addCardPopup = new PopupWithForm(
+  addCardPopupSelector,
+  handleAddCardSubmit
+);
+const editPopup = new PopupWithForm(editPopupSelector, handleProfileSubmit);
+const avatarEditPopup = new PopupWithForm(
+  avatarEditSelector,
+  handleAvatarSubmit
+);
+
+// api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "62fbc1de-42d6-41b7-8bb5-5b0736932e96",
+    "Content-Type": "application/json",
+  },
+});
+
+// promise.all
+api
+  .getCardAndUserInfo()
+  .then(([serverCardList, serverUserInfo]) => {
+    // load cards from server
+    cardList = new Section(
+      {
+        items: serverCardList,
+        renderer: (item) => {
+          const cardItem = creatCard(item);
+          cardList.addInitialItem(cardItem);
+        },
+      },
+      cardWrapperSelector
+    );
+    // add initial cards to DOM
+    cardList.renderItems(serverCardList);
+    //likeBtn active if isLiked=true
+
+    // load user bio from server
+    const name = serverUserInfo.name;
+    const job = serverUserInfo.about;
+    const avatar = serverUserInfo.avatar;
+    // add user bio to DOM
+    userProfile.setUserInfo(name, job);
+    userProfile.setAvatar(avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 // handler
+
 function handleAssignEditInput() {
   const userProfileData = userProfile.getUserInfo();
   editPopup.setInputValues(userProfileData);
 }
 
+function handleAvatarSubmit(inputObj) {
+  avatarEditPopup.showUploadingState();
+  const avatarLink = inputObj[avatarLinkInputId];
+  api
+    .updateProfileImage(avatarLink)
+    .then(avatarEditPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  avatarImg.src = avatarLink;
```

Comment:
Needs correction: please update the info and close the form in `then` block after the successful server response otherwise it will be updated even if an error occurred

Fix that everywhere

Reply:
Needs correction: 

1) please update the info and close the form in then block after the successful server response otherwise it will be updated even if an error occurred

Fix that everywhere

2) You should set the avatar using UserInfo class method setAvatar